python/utils/LinkedLibraryHolder.cpp

```diff
@@ -212,16 +246,32 @@ LinkedLibraryHolder::LinkedLibraryHolder() {
     }
   }
 
-  targets.emplace("qpp-cpu",
-                  RuntimeTarget{"qpp-cpu", "qpp", "default",
-                                "QPP-based CPU-only simulated QPU."});
+  // Set the default target
+  // If environment variable set with a valid value, use it
+  // Otherwise, if GPU(s) available, set default to 'nvidia', else to 'qpp-cpu'
+  defaultTarget = "qpp-cpu";
+  if (countGPUs() > 0) {
+    defaultTarget = "nvidia";
+  }
+  auto env = std::getenv("CUDAQ_DEFAULT_SIMULATOR");
+  if (NULL != env) {
```

Comment:
```suggestion
  if (env) {
```
While I like the Yoda expression, we don't really use `NULL` ...